### PR TITLE
Bugfix/delete refactor and fixes

### PR DESCRIPTION
**Pull request by katherinebernstein**

### CODE REVIEW

File: src/main/kotlin/com/openlattice/graph/Graph.kt [hunk 1]
```diff
@@ -487,6 +488,49 @@ class Graph(
         return neighbors
     }
 
+    override fun getEdgeEntitySetsConnectedToEntities(entitySetId: UUID, entityKeyIds: Set<UUID>): Set<UUID> {
+        val query = "SELECT DISTINCT ${EDGE_ENTITY_SET_ID.name} " +
+                "FROM ${E.name} " +
+                "WHERE (${SRC_ENTITY_SET_ID.name} = ? AND ${SRC_ENTITY_KEY_ID.name} = ANY(?) " +
```
Comment (you-leee) [inline src/main/kotlin/com/openlattice/graph/Graph.kt]: Just a minor thing, but can we use `SRC_IDS_SQL` and `DST_IDS_SQL` here?

File: src/main/kotlin/com/openlattice/data/DataGraphService.kt [hunk 2]
```diff
@@ -228,59 +196,6 @@ open class DataGraphService(
...
-
-        // delete entities
-        val entityWriteEvent = eds.deleteEntities(entitySetId, entityKeyIds, authorizedPropertyTypes)
-
-        logger.info("Deleted {} entities and {} vertices.", entityWriteEvent.numUpdates, verticesCount)
-
-        return entityWriteEvent
-    }
-
     override fun deleteAssociationsBatch(
```
Comment (you-leee) [inline src/main/kotlin/com/openlattice/data/DataGraphService.kt]: I don't think this is used anymore, right? (same for clearAssociationsBatch)

**Comment by makosblade** [comment]
If these are public APIs then we probably should update the docs as well

File: src/main/kotlin/com/openlattice/graph/Graph.kt [hunk 3]
```diff
@@ -487,6 +488,49 @@ class Graph(
...
+                "OR (${DST_ENTITY_SET_ID.name} = ? AND ${DST_ENTITY_KEY_ID.name} = ANY(?) )"
+
+        return PostgresIterable(
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/graph/Graph.kt]: I *think* pref BPGI b/c it's the Kotlin version ¯\_(ツ)_/¯

File: src/main/kotlin/com/openlattice/graph/Graph.kt [hunk 4]
```diff
@@ -487,6 +488,49 @@ class Graph(
...
+        ).toSet()
+    }
+
+
+    override fun getEdgeEntitySetsConnectedToEntitySet(entitySetId: UUID): Set<UUID> {
+        val query = "SELECT DISTINCT ${EDGE_ENTITY_SET_ID.name} " +
+                "FROM ${E.name} " +
+                "WHERE ${SRC_ENTITY_SET_ID.name} = ? OR ${DST_ENTITY_SET_ID.name} = ?"
+
+        return PostgresIterable(
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/graph/Graph.kt]: BPGI

File: src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+            entitySetId: UUID,
+            entityKeyIds: Set<UUID>,
+            deleteType: DeleteType,
+            principals: Set<Principal>
+    ): WriteEvent {
+        return clearOrDeleteEntities(entitySetId, entityKeyIds, deleteType, principals, skipAuthChecks = false)
+    }
+
+
+    // TODO rewrite this from scratch.
```
Comment (makosblade) [inline src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt]: 😢

File: src/main/kotlin/com/openlattice/graph/Graph.kt (hunk 1 above)
Comment (katherinebernstein) [reply src/main/kotlin/com/openlattice/graph/Graph.kt]: ooh true!

File: src/main/kotlin/com/openlattice/graph/Graph.kt (hunk 3 above)
Comment (katherinebernstein) [reply src/main/kotlin/com/openlattice/graph/Graph.kt]: +yes

File: src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+                .collect(Collectors.toSet())
+
+        if (unauthorizedAclKeys.isNotEmpty()) {
+            throw ForbiddenException("Unable to delete from entity set $entitySetId: missing required permissions " +
+                    "on associations for AclKeys $unauthorizedAclKeys")
+        }
+
+    }
+
+    private fun getAuthorizedPropertyTypesForDelete(
```
Comment (you-leee) [inline src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt]: Just a minor thing, but shouldn't we do the authorization checks in the controllers?

Comment (katherinebernstein) [reply src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt]: Theoretically yes, although the auth logic is pretty deep into the delete logic in multiple spots, and moving it into the controller would involve repeating a ton of logic in multiple places (which led to copy/paste and maintenance errors in our current implementation) as well as duplicating calls unnecessarily, so it seemed safer to me to add an auth-aware delete function in DataDeletionService.

Comment (you-leee) [reply src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt]: 👍

**Pull request merged**